sys/net/dsm/dsm.c

```diff
@@ -31,10 +31,18 @@ typedef struct {
     sock_dtls_session_t session;
     dsm_state_t state;
     uint32_t last_used_sec;
+    credman_tag_t tag;
+    credman_type_t type;
+#ifdef CONFIG_DTLS_ECC
+    unsigned char other_pub_x[DTLS_EC_KEY_SIZE];
+    unsigned char other_pub_y[DTLS_EC_KEY_SIZE];
+    size_t key_size;
+#endif
```

Comment:
I understand why you need to save the public key, but I believe this is not the place to do so, the storage should be done somewhere else (ideally we already know the public key we should be getting #20048 )